client/src/components/Dataset/DatasetView.vue

```diff
@@ -0,0 +1,373 @@
+    if (activeTab.value !== targetIndex) {
+        activeTab.value = targetIndex;
+    }
+}
+
+// Watch the 'tab' prop instead of the route path
+watch(() => props.tab, setActiveTabFromProp, { immediate: true });
+
+// Reset iframe loading when datasetId changes
+watch(
+    () => props.datasetId,
+    () => {
+        if (activeTab.value === TABS.PREVIEW) {
+            iframeLoading.value = true;
+        }
+    }
+);
+</script>
+<template>
+    <div v-if="dataset && !isLoading" class="dataset-view d-flex flex-column">
+        <header class="dataset-header flex-shrink-0">
+            <Heading h1 separator :collapse="headerState" class="dataset-name" @click="toggleHeader">
+                <div class="item-identifier">
+                    <span class="hid-box">{{ dataset.hid }}:</span>
+                    <span class="dataset-title">{{ dataset.name }}</span>
+                    <span v-if="contentState" class="state-pill ml-2" :class="`state-${dataset.state}`">
+                        <span v-if="hasStateIcon" class="state-icon mr-1">
+                            <FontAwesomeIcon fixed-width :icon="contentState.icon" :spin="contentState.spin" />
+                        </span>
+                        {{ stateText }}
+                    </span>
+                </div>
+            </Heading>
+
+            <div v-if="headerState === 'open'" class="header-details">
+                <div v-if="dataset.misc_blurb" class="blurb">
+                    <span class="value">{{ dataset.misc_blurb }}</span>
+                </div>
+                <span v-if="dataset.file_ext" class="datatype">
+                    <span v-localize class="prompt">format</span>
+                    <span class="value font-weight-bold">{{ dataset.file_ext }}</span>
+                </span>
+                <span v-if="dataset.genome_build" class="dbkey">
+                    <span v-localize class="prompt">database</span>
+                    <BLink class="value font-weight-bold" data-label="Database/Build" @click="onTabChange(TABS.EDIT)">{{
+                        dataset.genome_build
+                    }}</BLink>
+                </span>
+                <div v-if="dataset.misc_info" class="info">
+                    <span class="value">{{ dataset.misc_info }}</span>
+                </div>
+            </div>
+        </header>
+
+        <!-- Tab container - make it grow to fill remaining space and handle overflow -->
+        <div class="dataset-tabs-container flex-grow-1 overflow-hidden">
+            <!-- Make BTabs fill its container and use flex column layout internally -->
+            <BTabs pills card lazy class="h-100 d-flex flex-column" :value="activeTab" @input="onTabChange">
+                <BTab title="Preview" class="iframe-card">
+                    <div class="preview-container position-relative h-100">
+                        <!-- Loading indicator for iframe -->
+                        <div v-if="iframeLoading" class="iframe-loading">
```

Comment:
Can we use the LoadingSpan here instead of custom implementations? It ensures consistency across the user interface and promotes reusability.